runtime/integration-tests/src/generic/cases/lp/mod.rs

```diff
@@ -0,0 +1,912 @@
+// Copyright 2021 Centrifuge Foundation (centrifuge.io).
+//
+// This file is part of the Centrifuge chain project.
+// Centrifuge is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version (see http://www.gnu.org/licenses).
+// Centrifuge is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+use cfg_primitives::{Balance, CFG, SECONDS_PER_HOUR};
+use cfg_types::{
+	domain_address::Domain,
+	tokens::{CrossChainTransferability, CurrencyId, CustomMetadata},
+};
+use ethabi::{ethereum_types::U256, Token, Uint};
+use frame_support::{
+	assert_ok,
+	dispatch::RawOrigin,
+	pallet_prelude::ConstU32,
+	traits::{OriginTrait, PalletInfo},
+	BoundedVec,
+};
+use frame_system::pallet_prelude::OriginFor;
+use liquidity_pools_gateway_routers::{
+	AxelarEVMRouter, DomainRouter, EVMDomain, EVMRouter, FeeValues, MAX_AXELAR_EVM_CHAIN_SIZE,
+};
+use orml_traits::asset_registry::AssetMetadata;
+use pallet_evm::FeeCalculator;
+use sp_core::Get;
+use sp_runtime::traits::{BlakeTwo256, Hash};
+use xcm::v3::{
+	Junction::{AccountKey20, GlobalConsensus, PalletInstance},
+	Junctions::X3,
+	NetworkId,
+};
+
+use crate::{
+	generic::{
+		config::Runtime,
+		env::{Env, EvmEnv},
+		envs::runtime_env::RuntimeEnv,
+		utils::{genesis, genesis::Genesis, give_balance, last_event, ESSENTIAL},
+	},
+	utils::accounts::Keyring,
+};
+
+mod utils {}
+
+pub mod pool_management;
+
+const DEFAULT_BALANCE: Balance = 1_000_000;
+
+pub const DECIMALS_6: Balance = 1_000_000;
+pub const DECIMALS_18: Balance = 1_000_000_000_000_000_000;
+
+pub const USDC: CurrencyId = CurrencyId::ForeignAsset(100_001);
+pub const DAI: CurrencyId = CurrencyId::ForeignAsset(100_002);
+pub const FRAX: CurrencyId = CurrencyId::ForeignAsset(100_003);
```

Comment:
Maybe you can register these currencies with a `CurrencyInfo` in `generic/utils/currency.rs` to later use them more easily.

Reply:
Is this a genesis registration or just more easily calling registration?

Reply:
A more easy calling registration. The point with `CurrencyInfo` is that you have all related information associated with a currency and tied to that currency. So you do not need to have extra constants like `DECIMALS_X`, you can do `Frax::DECIMALS` or `Frax::ID` and things like that. Just less error prone in testing